src/transformStateWithClones.js

```diff
@@ -6,8 +6,42 @@
  *
  * @return {Object[]}
  */
+const addProperties = 'addProperties';
+const removeProperties = 'removeProperties';
+const clear = 'clear';
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let nextState;
+
+    switch (action.type) {
+      case clear:
+        nextState = {};
+        break;
+
+      case addProperties:
+        nextState = { ...currentState, ...action.extraData };
+        break;
+      case removeProperties:
+        nextState = { ...currentState };
+
+        for (const key of action.keysToRemove) {
+          delete nextState[key];
+        }
+        break;
+
+      default:
+        continue;
```

Comment:
The `default` case in the switch statement uses `continue`, which skips the current iteration of the loop without adding the `currentState` to the `states` array. If the intention is to ignore unknown actions, this is correct. However, if you want to record the state even when an unknown action is encountered, consider adding `currentState` to `states` before continuing.